pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/ServiceBusProcessor.kt

```diff
@@ -96,92 +104,164 @@ class ServiceBusProcessor {
+         *  @param invalidData MutableList<String>
+         *  @param createReportMessage CreateReportSBMessage
+         */
+        @JvmStatic
+        private fun SendToDeadLetter(invalidData:MutableList<String>, createReportMessage: CreateReportSBMessage){
+            if (invalidData.isNotEmpty()) {
+                //This should not run for unit tests
+                if (System.getProperty("isTestEnvironment") != "true") {
+                    // Write the content of the dead-letter reports to CosmosDb
+                    ReportManager().createDeadLetterReport(
+                        createReportMessage.uploadId,
+                        createReportMessage.dataStreamId,
+                        createReportMessage.dataStreamRoute,
+                        createReportMessage.stageName,
+                        createReportMessage.dispositionType,
+                        createReportMessage.contentType,
+                        createReportMessage.content,
+                        invalidData
+                    )
+                }
+                throw BadRequestException(invalidData.joinToString(separator = ","))
+            }
+        }
+        /**
+         *  Function to process the error by logging it and adding to the invalidData list and sending it to deadletter
+         *  @param reason String
+         *  @param invalidData MutableList<String>
+         *  @param createReportMessage CreateReportSBMessage
+         */
+        @JvmStatic
+        private fun ProcessError(reason:String, invalidData:MutableList<String>, createReportMessage: CreateReportSBMessage) {
+            LOGGER.error(reason)
+            invalidData.add(reason)
+            SendToDeadLetter(invalidData, createReportMessage)
+        }
+
+        /**
+         *  Function to validate the schema based on the schema file and the json contents passed into it
+         *  @param schemaFile String
+         *  @param objectMapper ObjectMapper
+
+         */
+        @JvmStatic
+        private fun ValidateSchema(jsonNode:JsonNode,schemaFile:File, objectMapper: ObjectMapper,invalidData:MutableList<String>,createReportMessage: CreateReportSBMessage) {
+            val schemaNode: JsonNode = objectMapper.readTree(schemaFile)
+            val schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7)
+            val schema: JsonSchema = schemaFactory.getSchema(schemaNode)
+            val schemaValidationMessages: Set<ValidationMessage> = schema.validate(jsonNode)
+
+            if (schemaValidationMessages.isEmpty()) {
+                LOGGER.info("JSON is valid against the content schema $schema.")
+            } else {
+               val reason ="JSON is invalid against the content schema $schema.Errors:"
+                schemaValidationMessages.forEach { invalidData.add(it.message) }
+                ProcessError(reason, invalidData,createReportMessage)
+            }
+        }
+        @JvmStatic
```

Comment:
I don't think we need to declare this @JvmStatic as it will only be used by Kotlin code.

Reply:
removed